components/style/values/specified/length.rs

```diff
@@ -220,14 +221,119 @@ impl CharacterWidth {
     }
 }
 
+/// Same as Gecko
+const ABSOLUTE_LENGTH_MAX: i32 = (1 << 30);
+const ABSOLUTE_LENGTH_MIN: i32 = - (1 << 30);
+
+/// Helper to convert a floating point length to application units
+fn to_au_round(length: CSSFloat, au_per_unit: CSSFloat) -> Au {
+    let au = (length * au_per_unit).round() as i32;
+    let clamped_au = min(max(au, ABSOLUTE_LENGTH_MIN), ABSOLUTE_LENGTH_MAX);
```

Comment:
Actually, I was wrong on this (sorry!). This needs to be clamped _before_ rounding and converting to integer, to avoid overflowing on the line before.